### ISSUE TITLE
exFAT FS -- incorrect 4GB max file size limit -- expected file size of 500gb -- kernel 5.10.63-v7l+ 

### ISSUE THREAD

**Issue by tonymet** [raspberrypi/linux#4795]
## Describe The Bug
exFat (using kernel module by samsung) reports incorrect filesize for files larger > 4gb.  `du -sh` = 3gb .  Actual file size is 500gb. 

When using fuse-exfat, the correct 500gb filesize is shown.  FS has also been tested on another system to verify file size.

## Steps to Reproduce
```
 # on another system, create > 4gb file on exfat partition
 sudo mount -t exfat -o ro  /dev/sda3 /mnt/photos
 du -h archive.tar
 stat archive2.tar
# make sure exfat module is used
mount|grep exfat
/dev/sda3 on /mnt/photos type exfat (ro,relatime,fmask=0022,dmask=0022,iocharset=utf8,errors=remount-ro)
```

## Expected behaviour
`du -h` = ~ 500gb
`stat` block size = 500gb / 512  = 980,969,822 blocks

## Actual behaviour
archive2.tar block size of `7891456 ` is  incorrect (*512 = 3.8gb -- wrong size) , inode size seems accurate (`502256548864`    )

**test A** -- du shows `3.8GB`
```
du -h archive2.tar
3.8G 3.8G    archive2.tar
```

**test B --** notice size = `502256548864    ` (correct),  blocks = `7891456 (3.8GB )` *incorrect*
```
stat archive2.tar
  File: archive2.tar
  Size: 502256548864    Blocks: 7891456    IO Block: 131072 regular file
Device: 803h/2051d      Inode: 939         Links: 1
Access: (0755/-rwxr-xr-x)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2021-12-30 08:08:34.000000000 -0800
Modify: 2021-12-30 02:09:16.590000000 -0800
Change: 2021-12-30 02:09:16.590000000 -0800
 Birth: 
Add a clear and concise description of what actually happened.
```

## System
* Which model of Raspberry Pi ? **Raspberry Pi 4 Model B Rev 1.2**
* Which OS and version (`cat /etc/rpi-issue`)? * 
``` 
$ cat /etc/rpi-issue
Raspberry Pi reference 2021-05-07
Generated using pi-gen, https://github.com/RPi-Distro/pi-gen, dcfd74d7d1fa293065ac6d565711e9ff891fe2b8, stage2

```

* Which firmware version (`vcgencmd version`)?
```
 vcgencmd version
Dec  1 2021 15:01:54
Copyright (c) 2012 Broadcom
version 71bd3109023a0c8575585ba87cbb374d2eeb038f (clean) (release) (start)
```

* Which kernel version (`uname -a`)?
```
uname -a
Linux pi4plus 5.10.63-v7l+ #1496 SMP Wed Dec 1 15:58:56 GMT 2021 armv7l GNU/Linux
```

**Logs**
none from dmesg

## Additional context
Add any other relevant context for the problem
```
 (start)
pi@pi4plus:/mnt/photos/Theresa-Personal-Macbook.local/2021-12-29 $ raspinfo|head -n 20
tee: raspinfo.txt: Read-only file system
System Information
------------------

Raspberry Pi 4 Model B Rev 1.2
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"

Raspberry Pi reference 2021-05-07
Generated using pi-gen, https://github.com/RPi-Distro/pi-gen, dcfd74d7d1fa293065ac6d565711e9ff891fe2b8, stage2

Linux pi4plus 5.10.63-v7l+ #1496 SMP Wed Dec 1 15:58:56 GMT 2021 armv7l GNU/Linux
Revision        : b03112
Serial          : 10000000908c0880
Model           : Raspberry Pi 4 Model B Rev 1.2
Throttled flag  : throttled=0x80000
Camera          : supported=0 detected=0
```

```
 uname -a
Linux pi4plus 5.10.63-v7l+ #1496 SMP Wed Dec 1 15:58:56 GMT 2021 armv7l GNU/Linux
```

### Modinfo exfat module
```
 modinfo exfat
filename:       /lib/modules/5.10.63-v7l+/kernel/fs/exfat/exfat.ko
author:         Samsung Electronics Co., Ltd.
description:    exFAT filesystem support
license:        GPL
alias:          fs-exfat
srcversion:     3CFFF6058657BA686C95591
depends:
intree:         Y
name:           exfat
vermagic:       5.10.63-v7l+ SMP mod_unload modversions ARMv7 p2v8
``` 

```
. cat /proc/cpuinfo
processor       : 0
model name      : ARMv7 Processor rev 3 (v7l)
BogoMIPS        : 108.00
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

[truncated]

Hardware        : BCM2711
Revision        : b03112
Serial          : 10000000908c0880
Model           : Raspberry Pi 4 Model B Rev 1.2
```

**Comment by tonymet** [comment]
Looks like this was a bug in upstream
https://github.com/torvalds/linux/commit/0c336d6e33f4bedc443404c89f43c91c8bd9ee11#diff-90577f7d4c49a024d9c212f277cc8bb8e79f76c8c4a2e9331905dc3813a454ff

I'll try to apply and test this patch.

**Comment by tonymet** [comment]
actually seems it was fixed in 284ad310542ffb4e5c2acf11d8a6420b5b7b9cea

I tested by building kernel branch rpi-5.10.y @ d261fd9f9 and the issue was resolved. thanks
